### ISSUE TITLE
The question about directed and undirected graphs

### ISSUE THREAD

**Issue by hust-nj** [graphdeeplearning/benchmarking-gnns#22]
Hi, I see your code in
https://github.com/graphdeeplearning/benchmarking-gnns/blob/92c762a5a8713f5804c69c1808128a6b3e8f4535/data/superpixels.py#L114-L144

, find the superpixel graph is directed graph not an undirected graph? Isn't it true?
I see **Figure 1** of your paper, you draw MNIST and CIFAR10  as undirected graphs, but it should be directed graphs based on your code?

**Comment by chaitjo** [comment]
Hello, no, we use undirected graphs for superpixels. I suggest you run the code and provide output to support your conclusion of the graphs being directed. 

In general, according to DGL, all graphs are directed (https://docs.dgl.ai/api/python/graph.html#dgl.DGLGraph). To make an undirected graph (like in our case), one must add two edges (bi-directional) between each pair of nodes we'd like to connect. I believe our code implements this: https://github.com/graphdeeplearning/benchmarking-gnns/blob/92c762a5a8713f5804c69c1808128a6b3e8f4535/data/superpixels.py#L133 

(But I would be happy to work on a fix if you can demonstrate that it does not...)

**Comment by hust-nj** [comment]
Yes, I verified it additionally, I insert the code
```python
    print(torch.unique(trainset[0][0].all_edges()[0], sorted=True, return_counts=True))
    print(torch.unique(trainset[0][0].all_edges()[1], sorted=True, return_counts=True))
    exit()
```
after this line 
https://github.com/graphdeeplearning/benchmarking-gnns/blob/92c762a5a8713f5804c69c1808128a6b3e8f4535/main_superpixels_graph_classification.py#L104
Find that every node with out-degree 8 but not all in-degree 8
Can you **verify** this ?
Thank you

**Comment by chaitjo** [comment]
Will check and get back to you. Cc. @vijaydwivedi75 

What might be happening: A node is an 'outlier', i.e. very far away from a central cluster of nodes. Now, for this outlier node, we would have k nearest neighbors, as usual. However, it will never/seldom be among the k nearest neighbors for any of the other nodes.

**Comment by chaitjo** [comment]
Indeed, we found the superpixel graphs to be directed, @hust-nj thanks for pointing this out. However, we decided to continue with MNIST and CIFAR10 as directed graphs in our updated version of the paper as we felt these datasets are more of a 'sanity check' for all GNNs. Additionally, as the task being studied is graph classification, we are assuming that directed vs. undirected graphs should not significantly impact performance, as the model takes a mean over all node representations to represent the graph.

Do let us know if this makes sense or if you have alternate results.
